### PR TITLE
fix: 🐛 aliasing bugs

### PR DESCRIPTION
**Pull request by DhariniJeeva**
:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-13525)

## Description

This pr fixes a few things - 
1. Hides alias sidebar container when there are no associated aliases in DC 
2. Hides target ID from alias form creation/update within a target
3. Adds confirmation messages for delete and clear aliases 

## Screenshots (if appropriate):

 read-only in new form
t1

editable in edit state
t2

## How to Test

:technologist: [Admin preview](PATH_TO_FEATURE)

:desktop_computer: [Desktop preview](PATH_TO_FEATURE)

## Checklist:

- [ ] I have added before and after screenshots for UI changes
- [ ] I have added JSON response output for API changes
- [ ] I have added steps to reproduce and test for bug fixes in the description
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: ui/admin/app/templates/scopes/scope/targets/target/manage-alias.hbs [hunk 1]
```diff
@@ -48,6 +48,7 @@
   <page.body>
     <Form::Alias
       @model={{@model}}
+      @hideTargetId='true'
```
Comment (DhariniJeeva) [inline ui/admin/app/templates/scopes/scope/targets/target/manage-alias.hbs:51]: waiting on feedback for this, not sure if we should still hide the target id in edit state

Comment (DhariniJeeva) [reply ui/admin/app/templates/scopes/scope/targets/target/manage-alias.hbs:51]: this has been changed to readonly in new and editable in edit view, see [thread](https://hashicorp.slack.com/archives/C012GTH1C4E/p1713474813278209) here

File: addons/core/translations/en-us.yaml [hunk 2]
```diff
@@ -60,6 +60,7 @@ questions:
   remove-confirm: Are you sure you want to remove this association?
   abandon-confirm: Are you sure you want to navigate away?  Your unsaved changes will be lost.
   retry-confirm: Would you like to retry?
+  clear-confirm: Are you sure you want to clear the resource id?
```
Comment (ZedLi) [inline addons/core/translations/en-us.yaml:63]: I get a second modal that pops up for a fraction of a second when I hit clear:


https://github.com/hashicorp/boundary-ui/assets/5783847/b13d3cb7-6a1a-4d69-8af6-8934e8b26ee2

**Review by ZedLi — CHANGES_REQUESTED** [review]
Was it also intentional to not have a manage dropdown in the alias details page? I can only delete from the list view

**Comment by DhariniJeeva** [comment]
> Was it also intentional to not have a manage dropdown in the alias details page? I can only delete from the list view

ah I missed that, will add

File: addons/core/translations/en-us.yaml (hunk 2 above)
Comment (DhariniJeeva) [reply addons/core/translations/en-us.yaml:63]: thanks! updated this

File: ui/admin/app/templates/scopes/scope/aliases/alias/index.hbs [hunk 3]
```diff
@@ -23,12 +23,76 @@
...
+        <dd.ToggleButton
+          data-test-manage-dropdown
+          @text={{t 'actions.manage'}}
+          @color='secondary'
+        />
+        <dd.Interactive
+          data-test-manage-alias-dropdown-delete
+          @color='critical'
+          @text={{t 'actions.delete'}}
+          {{on 'click' (fn this.aliases.toggleModal @model)}}
```
Comment (ZedLi) [inline ui/admin/app/templates/scopes/scope/aliases/alias/index.hbs:38]: Hmm the one from managed targets has two dropdown options for both clear and delete, how come these ones are different and both options are in a modal when you "delete"?

File: ui/admin/app/templates/scopes/scope/aliases/alias/index.hbs [hunk 4]
```diff
@@ -23,12 +23,76 @@
       </nav.link>
     </Rose::Nav::Tabs>
   </page.navigation>
-
+  {{#if (can 'delete model' @model)}}
```
Comment (ZedLi) [inline ui/admin/app/templates/scopes/scope/aliases/alias/index.hbs:26]: Do you need delete permissions if you just want to clear?

File: ui/admin/app/controllers/scopes/scope/aliases/index.js [hunk 5]
```diff
@@ -73,11 +69,10 @@ export default class ScopesScopeAliasesIndexController extends Controller {
   @notifyError(({ message }) => message, { catch: true })
   @notifySuccess('notifications.clear-success')
   async clearAlias(alias) {
```
Comment (ZedLi) [inline ui/admin/app/controllers/scopes/scope/aliases/index.js:71]: I notice we have a route action `clearAlias` and `deleteAlias` as well as controller ones? Did we need multiple versions that do almost the same thing?

Comment (DhariniJeeva) [reply ui/admin/app/controllers/scopes/scope/aliases/index.js:71]: yes we need as the other ones are for the aliases within target route

Comment (ZedLi) [reply ui/admin/app/controllers/scopes/scope/aliases/index.js:71]: We couldn't just share it? Since you can inject controllers?

Comment (DhariniJeeva) [reply ui/admin/app/controllers/scopes/scope/aliases/index.js:71]: Can we share controllers from one route to another? 🤔

Comment (ZedLi) [reply ui/admin/app/controllers/scopes/scope/aliases/index.js:71]: I think technically yes but we can talk about this offline in tech time, this isn't blocking

File: ui/admin/app/templates/scopes/scope/aliases/alias/index.hbs (hunk 4 above)
Comment (DhariniJeeva) [reply ui/admin/app/templates/scopes/scope/aliases/alias/index.hbs:26]: uh... I thought we did, checking with BE

**Comment by calcaide** [comment]
I tested deleting an alias from Aliases list and from the targets > target detail > Alias side bar > manage > Alias detail> manage > delete and the confirmation modals are different, the design and the content. Is this expected?

Modal from Aliases domain:
Screenshot 2024-04-19 at 2 04 50 PM

Modal from Targets domain:
Screenshot 2024-04-19 at 2 04 57 PM

**Comment by DhariniJeeva** [comment]
> I tested deleting an alias from Aliases domain (list and alias domain) and from the targets domain, targets list > target detail > Alias side bar > manage > Alias detail> manage > delete and the confirmation modals are different, the design and the content. Is this expected?
> 
> Modal from Aliases domain: Screenshot 2024-04-19 at 2 04 50 PM
> 
> Modal from Targets domain: Screenshot 2024-04-19 at 2 04 57 PM

yes we will be getting rid of the modal due to permission issues, see this thread [here](https://hashicorp.slack.com/archives/C06NK4E647J/p1713547552640849) for more details.

File: ui/admin/app/templates/scopes/scope/aliases/alias/index.hbs [hunk 6]
```diff
@@ -23,12 +23,76 @@
...
+    <page.actions>
+      <Hds::Dropdown as |dd|>
+        <dd.ToggleButton
+          data-test-manage-dropdown
+          @text={{t 'actions.manage'}}
+          @color='secondary'
+        />
+        <dd.Interactive
+          data-test-manage-alias-dropdown-delete
```
Comment (calcaide) [inline ui/admin/app/templates/scopes/scope/aliases/alias/index.hbs:35]: None blocking comment: are we planning on using these `data-test-*` selectors for tests?

Comment (DhariniJeeva) [reply ui/admin/app/templates/scopes/scope/aliases/alias/index.hbs]: thanks, we don't need this, removed

File: ui/admin/app/templates/scopes/scope/aliases/alias/index.hbs (hunk 4 above)
Comment (DhariniJeeva) [reply ui/admin/app/templates/scopes/scope/aliases/alias/index.hbs]: discussed offline, we don't need del permissions to clear, we removed the modal and listed the actions now

File: ui/admin/app/templates/scopes/scope/aliases/index.hbs [hunk 7]
```diff
@@ -48,11 +48,11 @@
                   @route='scopes.scope.aliases.alias'
                   @model={{B.data.alias.id}}
                 >
-                  {{B.data.alias.displayName}}
+                  {{B.data.alias.value}}
```
Comment (DhariniJeeva) [inline ui/admin/app/templates/scopes/scope/aliases/index.hbs:51]: this should be value instead of name

Comment (DhariniJeeva) [reply ui/admin/app/templates/scopes/scope/aliases/index.hbs:51]: this is based on design/product [feedback](https://hashicorp.slack.com/archives/C012GTH1C4E/p1713560858991069). Value is a required field and we show that in the table as it makes more sense than ID for aliases

**Review by cameronperera — APPROVED** [review]
LGTM

File: ui/admin/app/templates/scopes/scope/aliases/alias/index.hbs [hunk 8]
```diff
@@ -23,7 +23,36 @@
...
+      canRead=(can 'read model' @model) canDelete=(can 'delete model' @model)
+    )
+    as |perms|
+  }}
+    <page.actions>
 
+      {{#if (or perms.canRead perms.canDelete)}}
+        <Hds::Dropdown as |dd|>
+          <dd.ToggleButton @text={{t 'actions.manage'}} @color='secondary' />
+          {{#if (and perms.canRead @model.destination_id)}}
```
Comment (ZedLi) [inline ui/admin/app/templates/scopes/scope/aliases/alias/index.hbs:37]: Shouldn't we be checking update permissions?

**Review by calcaide — APPROVED** [review]
LGTM, great bugs fixing 🧰 🙌

File: ui/admin/app/templates/scopes/scope/aliases/alias/index.hbs (hunk 8 above)
Comment (DhariniJeeva) [reply ui/admin/app/templates/scopes/scope/aliases/alias/index.hbs:37]: actually I wasn't sure, our `canUpdate` is bundled with the `canSave` in base [model](https://github.com/hashicorp/boundary-ui/blob/f65f483611105ac2735fda075a4184a77cda293d/addons/api/addon/abilities/model.js#L46) ability and we normally check for `read` before letting the user click on a resource to edit.  so I checked for canRead.  But in this case, the user wouldn't hit save so we should check for `canUpdate`?

Comment (ZedLi) [reply ui/admin/app/templates/scopes/scope/aliases/alias/index.hbs:37]: Right, normally we'd use `canSave` from edit view but in this case you could actually update it without read permissions straight from the list view, you could just have edit permissions for whatever reason and just clear it without even reading the alias. We should obviously test this but I would expect us to be just checking `canUpdate`

File: ui/admin/tests/acceptance/aliases/update-test.js [hunk 9]
```diff
@@ -78,4 +93,21 @@ module('Acceptance | aliases | update', function (hooks) {
     assert.dom(NAME_FIELD_SELECTOR).hasValue(`${name}`);
     assert.strictEqual(instances.alias.name, name);
   });
+
+  test('users have the option to clear an alias', async function (assert) {
```
Comment (ZedLi) [inline ui/admin/tests/acceptance/aliases/update-test.js:97]: It probably would be good to have a quick test that the user can't clear without update permissions but this is not blocking

**Review by ZedLi — APPROVED** [review]
Looks great! Thanks for the fixes

File: ui/admin/tests/acceptance/aliases/update-test.js (hunk 9 above)
Comment (DhariniJeeva) [reply ui/admin/tests/acceptance/aliases/update-test.js:97]: agree, added!

**Pull request merged**